src/pystack/_pystack/elf_common.cpp

```diff
@@ -144,13 +118,128 @@ CoreFileAnalyzer::~CoreFileAnalyzer()
     close(d_fd);
 }
 
+void
+CoreFileAnalyzer::removeModuleIf(std::function<bool(Dwfl_Module*)> predicate)
```

Comment:
This can be made `const`

```suggestion
CoreFileAnalyzer::removeModuleIf(std::function<bool(Dwfl_Module*)> predicate) const
```

you need to update the header as well.